### ISSUE TITLE
mindLAMP App Translation

### ISSUE THREAD

**Issue by ertjlane** [BIDMCDigitalPsychiatry/LAMP-platform#697]
Introducing translated versions of mindLAMP

mindLAMP Translations in French, German, Danish, Italian, Spanish, Mandarin, & Korean. Translations are provided by independent translation bureaus, and simply need to be coded into the platform.

**Comment by ertjlane** [comment]
Hello, I've attached here an additional set of translations, supplemental to those that have already been provided via email.
[Archive 2.zip](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/9651047/Archive.2.zip)

**Comment by sarithapillai8** [comment]
We already have Spanish translation file. So do we need to replace the current one with this newly provided one?
@ertjlane

**Comment by ertjlane** [comment]
@sarithapillai8 I suspect the Spanish translation file is identical to the one I shared in the comment here. Could you check and make sure, and if not please send me the file? 

Also  what is the ETA we can expect for these translations to be instantiated? Last I spoke to the team it was two weeks, but we would like if this first round was done sooner - though we are expecting another set of translations to come in this Friday, so there will be another round of uploads after the deadline you provide for this first batch.

**Comment by sarithapillai8** [comment]
Please find the current file here - [es.zip](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/9671025/es.zip)
The file contents seems different.  

It seems many more translations needed for completing this translation. We hope more strings will be provided. Do you want to update dashboard staging partially with the provided translations? so that you can check and update the files accordingly. Please let us know.
@ertjlane

**Comment by ertjlane** [comment]
For the Spanish translation it does appear that there is some overlap, but yes they have different content. Would it be possible for your team to unify the content of the two files, since you are already doing a partial translation?

There will be another set of translations coming which will add to this as well. And yes, updating the dashboard with the partial translations for now would be fantastic, thank you!

**Comment by ertjlane** [comment]
Thanks for checking! 

Thank you for updating the dashboard, and promptly! That is noteworthy that only a small portion of the strings seem to be translated, since three different people have been through the mindLAMP dashboard copying text to assemble these translations, and the majority of them should have been covered by the 'old_text' files. Nothing your team did wrong, it's just striking to me. And to make sure I understand, each language should have only one .json file with all translations, correct? So for any incoming updates I should assemble the files already sent and append the content to the update.

**Comment by ertjlane** [comment]
If it's possible, could you attach the translated Spanish file (with the english left for those parts that are missing), and the original only English string file, so I can see exactly how much translated material is still missing?

**Comment by sarithapillai8** [comment]
> each language should have only one .json file with all translations, correct? So for any incoming updates I should assemble the files already sent and append the content to the update.

Yes.

**Comment by ertjlane** [comment]
Thanks for this, I'll use these as a frame of reference and a final check. We have received updated files for Danish (completely new), Korean and German. We are currently waiting on further updates from Chinese, Spanish and Italian. 
I've attached updates for all languages here, with all previous translations included in the new files as you specified. Please implement these asap, and once that is done I'll put the issue on hold until we receive the next round of translations. Let me know if you have any questions or thoughts! And thank you for the prudence taken here! 

[Translations Round 2.zip](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/9716859/Translations.Round.2.zip)

**Comment by sarithapillai8** [comment]
Updated dashboard-staging with these new set of files. 

You can see the language texts in login screen. 
![image](https://user-images.githubusercontent.com/63710347/194222263-89dc7c51-47fd-4916-8e83-84625fadbdbb.png)
Please have a look and let me know if there is any mistake in matching the language code. 
Currently using the following ones :
"en-US", "es-ES", "hi-IN", "de-DE", "da-DK", "fr-CA", "ko-KR", "it-IT", "zh-CN"

You can find these codes here - https://github.com/BIDMCDigitalPsychiatry/LAMP-dashboard/blob/master/src/locale_map.json. If any different code is expected, please let me know. Thank you.

**Comment by ertjlane** [comment]
Thanks for doing this! I can see the languages have been implemented. Something odd is happening though, certain pieces of text are appearing as English, despite being implemented in the translation file. For example, in Danish 'Please select the study' has been translated, and I can see the Danish translation in the .json file on github. But I see the text in English on the dashboard. I've included screenshots below. Do you have any sense why this might be occurring?

<img width="1286" alt="Screen Shot 2022-10-06 at 12 02 07 PM (2)" src="https://user-images.githubusercontent.com/104088715/194362822-d0a47fab-89fc-46d2-a975-4a0c1abbd9d2.png">
<img width="1287" alt="Screen Shot 2022-10-06 at 12 01 09 PM (2)" src="https://user-images.githubusercontent.com/104088715/194362851-d1e28ec1-3cd4-4e67-969a-ab7f57a3bf4c.png">

**Comment by sarithapillai8** [comment]
@ertjlane 
We need to use the exact same string in json file. Each and every special characters and follow the upper/lower cases.
 "Please select the Study." - the '.' is missing as key in json.  Also for "Please select the sensor spec.", the json having 'Please select sensor spec'. As already a team is working on this, could you please update these?

**Comment by ertjlane** [comment]
@sarithapillai8 Thanks for the follow up. So I looked into this, and you're right that there is some incoherence between English keys. However, there still appear to be several keys that are perfectly matched to the dashboard text, yet the translation still doesn't appear. This happens across languages (though not for all text) as demonstrated in the text below. Could you look into this while I compile adjustments for the incorrect keys?
<img width="245" alt="Screen Shot 2022-10-11 at 11 55 11 AM (2)" src="https://user-images.githubusercontent.com/104088715/195194029-74eb98a4-9763-4c28-8e18-5864ef2677ba.png">
<img width="581" alt="Screen Shot 2022-10-11 at 11 55 14 AM (2)" src="https://user-images.githubusercontent.com/104088715/195194038-fd467bed-6e1a-478c-81e4-eac3aeffa19a.png">
<img width="303" alt="Screen Shot 2022-10-11 at 12 09 06 PM (2)" src="https://user-images.githubusercontent.com/104088715/195194209-fab98455-da80-4d84-b415-254aeb11fce5.png">
<img width="203" alt="Screen Shot 2022-10-11 at 12 09 08 PM (2)" src="https://user-images.githubusercontent.com/104088715/195194225-05846b8a-8967-4c89-9d52-5be6cd29f1c6.png">

**Comment by sarithapillai8** [comment]
@ertjlane For Search text, the actual text required is just Search without dots. It is an exceptional case. 2-3 strings were missing translation function in code.  I have added those and updated in dashboard-staging. If you spot more, please let me know. I can check from here also.

**Comment by ertjlane** [comment]
I'm going through the translations checking for errors. I'm noticing that the core English language (the key) is different between languages. So for example, English text in the exact same place in the Dashboard will say different things (ex: Study instead of Group). Can you tell me what the reason for this is? 
<img width="411" alt="Screen Shot 2022-10-13 at 4 19 13 PM" src="https://user-images.githubusercontent.com/104088715/195702471-c364a3b3-1f1e-4090-8886-777ac782fc72.png">
<img width="403" alt="Screen Shot 2022-10-13 at 4 19 13 PM (2)" src="https://user-images.githubusercontent.com/104088715/195702501-d6547312-4e32-4173-ba64-9a38eb0902d5.png">

**Comment by sarithapillai8** [comment]
@ertjlane 
This seems a bug. Study was renamed as Group a few months back. English translation files contain key with Study , but value with Group. That is why it is correct with English language selection. I can change code-base, but can you please assure that translation files don't contain Study instead of Group in new set. I can change the current files. Also could you please confirm that we need to text Group instead of Study in every place including the Researcher menu title?

**Comment by ertjlane** [comment]
I've gone through the dashboard and this set of translations, there are some cases where study still exists in the main English dashboard. For the time being, I have noted in the translations where the error exists, and where the 'Study' needs to be adjusted to group. I've attached here a spreadsheet of errors for German, Korean, Danish and French. I've listed the errors resulting from incorrectly written keys, and those where the error seems to be something else (listed as 'unknown'. Let me know if you have any questions about the material here.
[Translation Errors - 10_19_2022 - EL (1).xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/9824538/Translation.Errors.-.10_19_2022.-.EL.1.xlsx)

**Comment by ertjlane** [comment]
Hello all. I've attached here two new, full translation documents, for Spanish and Italian. I've checked each for the key errors that emerged in the previous languages. In addition, I've attached a list of errors that occurred in the other translations for unknown reasons, so please check if those same errors have emerged in these languages (disregard the green highlights). Thank you, and please let me know that you're working on this and addressing the errors noted in the above comment! 
[Archive 3.zip](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/9842275/Archive.3.zip)

**Comment by sarithapillai8** [comment]
We are working on these and will update ASAP.

**Comment by sarithapillai8** [comment]
@ertjlane 
We have updated many strings in both code base and json files to help the translation. 

Could you please confirm whether we need to do translation in Data portal section? It was not implemented by our team and no translation was implemented there. I have added some, but could you please let me know if we need to do all the strings there? 

Also is it possible to consider the latest translation files from the repo? Because I have corrected some strings in json files - https://github.com/BIDMCDigitalPsychiatry/LAMP-dashboard/tree/master/public/locales.

**Comment by sarithapillai8** [comment]
The below string is not correct in translation. Please use separate translation for them : 

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SARITH~1.PIL/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SARITH~1.PIL/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#1155CC" vlink="#1155CC">



"There   are no Assess activities available" | "Es sind keine   Beurteilungs-aktivitäten verfügbar." | Key Error | The translation   given is not correct : "There are no Assess/Manage/Learn/scheduled   activities available.": "\uc0ac\uc6a9 \uac00\ub2a5\ud55c   \ud3c9\uac00/\uad00\ub9ac/\ud559\uc2b5/\uc608\uc57d\ub41c \ud65c\ub3d9\uc774   \uc5c6\uc2b5\ub2c8\ub2e4.",
-- | -- | -- | --
"There   are no Manage activities available" | "Es sind keine   Management-aktivitäten verfügbar." |  
"There   are no Learn activities available" | "Es sind keine   Lern-aktivitäten verfügbar." |  
"There   are no scheduled activities available" | "Es sind keine   Planungsaktivitäten verfügbar." |  



</body>

</html>

**Comment by ertjlane** [comment]
Thanks for looking at this! Yes, the data portal section strings NEED to be implemented. Thanks for catching the ones shared above, if you could implement the corrected ones you've provided that would be great!

Could you please clarify what you mean by 'Also is it possible to consider the latest translation files from the repo'?

**Comment by sarithapillai8** [comment]
>Could you please clarify what you mean by 'Also is it possible to consider the latest translation files from the repo'?

I have updated some strings in the translation files and updated in the repo. So when updating new strings to that, could you consider those changes? You can see the latest files in the link given above.

**Comment by sarithapillai8** [comment]
[Translations.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/9889549/Translations.xlsx)

We have implemented translations in Data portal section. Please find the strings to be used in the attached sheet. Please use the exact same strings.

**Comment by ertjlane** [comment]
[Data Portal Translations.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/9933436/Data.Portal.Translations.xlsx)
I've attached the updated data portal translations here.

**Comment by sarithapillai8** [comment]
@ertjlane Translation files were given as json file before. Chinese version was given in a different text format also. Could you please provide similar to previous versions?

**Comment by ertjlane** [comment]
@sarithapillai8 It was not made clear that the excel file was expected to be transitioned into a .json. I understand that transitioning the file is tedious. However, up to this point the burden of reformatting the files has fallen entirely on me, which has taken a considerable amount of time. This is a high priority project, and I believe it would proceed more quickly if the reformatting was divided among members of your team so as to keep things moving at a quicker pace. If the task is broken among several people I do not imagine it will take each too long. I understand that you did this for the Hindi translation, and in the interest of moving this more quickly I would ask that your team take it on, irritating and time consuming as it may be.

**Comment by ertjlane** [comment]
Following up, I just checked through the staging dashboard, and it appears that a number of the errors that I passed on over a week ago that still have not been addressed. I understand I did not ask for a specific ETA on this, but can the error review/fix please be completed as soon as possible? Particularly French, as we have collaborators expecting to use it soon. I see do that many of the errors have been addressed.

**Comment by sarithapillai8** [comment]
@ertjlane I understand the difficulties with translation file updates. I hope we can divide the work. But can you take care of the Chinese version?
I can double check the bugs missed give more priority to French.

**Comment by sarithapillai8** [comment]
@ertjlane We have updated French version fixes in dashboard-staging including data portal changes. QA testing is in progress. Still you can check that in parallel if you want to.

**Comment by Kaygijzen** [comment]
@ertjlane Did you forget to add the Dutch translations file? I couldn't find it.

**Comment by ertjlane** [comment]
@sarithapillai8 Thank you, yes I can reformat the chinese data portal materials I sent you. I just looked over the french document. Much has been improved, and I appreciate that! I did notice that there are certain texts implemented in the french translation file in github (ex: the months of the year, filter, collapse), which don't appear in the staging dashboard despite being loaded into github. Do you know what the reason for this might be? 

@Kaygijzen Hello Kay! I'm not quite sure what you're referring to. mindLAMP has no Dutch translation, nor is there any existing effort to implement as such.

**Comment by Kaygijzen** [comment]
@ertjlane My bad. I thought there was, since you listed Dutch among the translations in the top comment in this thread.

**Comment by sarithapillai8** [comment]
@ertjlane Not sure about what are saying. If the changes are in master branch, then it always appears on dashboard-staging. I took the update from your latest changes and added some more changes and updated dashboard-staging.

**Comment by ertjlane** [comment]
Chinese data portal materials .json file: 
[Chinese dataportal.json.zip](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/9974851/Chinese.dataportal.json.zip)

**Comment by ertjlane** [comment]
For the sake of expediency I'm moving this up to urgent priority, as it effects collaborators. Just let me know the soonest the errors I submitted can be addressed in their entirety.

**Comment by sarithapillai8** [comment]
@ertjlane We are going to start with integrate Italian version for data portal. We both worked in Danish version. So to avoid that, could you inform which one you are currently working on?
We can update a build with error fixes possible with French version and can add Chinese version to dashboard-staging today.

**Comment by ertjlane** [comment]
@sarithapillai8 For the time being I do not have any more materials to add. I am expecting some German fix translations to be returned, which I will upload, and I'll give you a heads up before I do that. Great to hear about the French and Chinese update, is there a timeline for the errors submitted in the other languages to be addressed? I looked at dashboard staging yesterday and saw there was still some missing materials.

**Comment by ertjlane** [comment]
@sarithapillai8 I see some updates have been pushed, thank you for that. Can we get an ETA for the rest of the errors to be fully addressed?

**Comment by sarithapillai8** [comment]
@ertjlane We are trying to update everything ASAP. As these are manual changes, it is taking time and it is really hard to say an ETA on this. Most of the errors are addressed already and we started going through each language in detail. Whatever missing translations found, we will be updating them in different language files  and will post here to check whether to keep them as they are or you can provide translations for them. When correcting errors in one language, we will be correcting the same in all other ones. I hope, by the end of this week we can update an ETA on this.

**Comment by sarithapillai8** [comment]
We have updated some more fixes in dashboard-staging. Please find the missing translations of German language in the attachment. Also please go through different activity settings and most of them were missing in given translation list.
[Missing-Translations.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10029171/Missing-Translations.xlsx)

**Comment by ertjlane** [comment]
Thank you for this! A priority list for languages is French (I know close to completion if not done), Italian, Spanish, German, Chinese, Danish & Korean.

**Comment by divyav2020** [comment]
We have updated dashboard staging with language translations for different languages. Attaching the file with missing translations found.


Note : Some of the translations are not getting updated on the website even we have updated them in code. We are trying to figure it out.

We have updated the following language strings: French, German, Spanish, Korean and Chinese

We will be continuing QA and further changes will be updated in coming weeks.

Note: Latest changes are not updated in dashboard staging. Build getting failed. We will be checking the issue.

[Missing Translations.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10094200/Missing.Translations.xlsx)

**Comment by sarithapillai8** [comment]
>Note : Some of the translations are not getting updated on the website even we have updated them in code. We are trying to figure it out. 

>Note: Latest changes are not updated in dashboard staging. Build getting failed. We will be checking the issue.

The issues mentioned above is fixed and updated in dashboard-staging

**Comment by ertjlane** [comment]
This is great news. I am beginning to receive translations for the missing materials. I can put in the translations for the majority of the dashboard myself, but for data portal translations how should I implement those? Is there a way for me to input those as with other translations, or should I send them to you in a .json format?

**Comment by sarithapillai8** [comment]
You can send me the translation file as our team already working on it. Otherwise when we update from here, there is a chance of missing some.

**Comment by ertjlane** [comment]
Understood, I've attached a data portal update for the German translation. 

We have also just had the French translation reviewed by French mindLAMP users, with just a few small notes. They pointed initials of the week in the feed section do not represent the French week initials (which we didn't think of). The initials are 'D L M M J V S' respectively. I'm not sure how one would implement such a thing so I'm raising it here. They also noted that some of the iphone pop ups were not translated - are these even in our wheelhouse to translate, or is that an issue with say the apple iphone translation? References for each of these issues can be found in the attached pdf of feedback on the 4th (initials) and last (iphone popups) pages. Overall the feedback was very minimal, so the fact that we're just cleaning up these last pieces is great. Thank you. 
[Translation Check mindlamp.pdf](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10134376/Translation.Check.mindlamp.pdf)

[Update_German_DP_1122.json.zip](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10134399/Update_German_DP_1122.json.zip)

**Comment by sarithapillai8** [comment]
> We have also just had the French translation reviewed by French mindLAMP users, with just a few small notes. They pointed initials of the week in the feed section do not represent the French week initials (which we didn't think of). The initials are 'D L M M J V S' respectively. I'm not sure how one would implement such a thing so I'm raising it here

We have implemeted this and updated some of the other fixes in dashboard-staging.

**Comment by ertjlane** [comment]
Great! Do you have any feedback regarding the iphone pop ups?

**Comment by sarithapillai8** [comment]
>Do you have any feedback regarding the iphone pop ups?

We are working on it. We will update ASAP.

**Comment by sarithapillai8** [comment]
@ertjlane 
Please find the attached document for missing translations requires in app - 
[mindLAMP2_MobileApp_Localization.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10154724/mindLAMP2_MobileApp_Localization.xlsx)

**Comment by ertjlane** [comment]
Excellent, thank you!

**Comment by sarithapillai8** [comment]
@ertjlane We are about to finish the feedback items along with QA fixes. So hopefully we can finish everything except the app side changes by the end of next week.  In between hope you will be providing the translations for that in the given excel sheet given here - https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/issues/697#issuecomment-1337410107

**Comment by jijopulikkottil** [comment]
@ertjlane There are few more strings and those are added to the excel sheet. Please use this updated file.
[mindLAMP2_MobileApp_Localization.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10177010/mindLAMP2_MobileApp_Localization.xlsx)

**Comment by sarithapillai8** [comment]
@ertjlane 
The translations the collaborator given in the document are added and updated in staging. As you noticed, some of them are not actually seems like an issue.

**Comment by sarithapillai8** [comment]
> There was a comment in the Italian feedback sheet  - PORTALE: 
When I start the Survey task it appears for each question:
Domanda numero su un totale di
But it should be Question …x… on a total number of ….:
Domanda “X” su numero su un totale di 30

Actually it is showing like  - " Domanda 1 di 7", do we really need to update this? We are translating "Question 1 of 7" in this case.

**Comment by divyav2020** [comment]
The translations given in the document are added and updated in staging. 
We are waiting for the missing translations

**Comment by sarithapillai8** [comment]
App side is also waiting for the translations provided above. So we have removed QA testing status.

**Comment by jijopulikkottil** [comment]
For mobile app, hope we don't need to localise App Store information and app display name.

**Comment by ertjlane** [comment]
Hello all,

Thank you for your work on this, great to hear everything is updated! In terms of the timeline, I have received an app translation from one of the sites I sent it to (each language was sent to a different group), but had to send it back due to missing information. I will reach out to the sites that have not contacted me by Friday asking for an update, and the same for the missing dashboard translations. So at this point we're mostly operating on standby. 

I've attached some small updates for the Italian translation here. 
[Italian Translation Update-12-13-22.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10219412/Italian.Translation.Update-12-13-22.xlsx)

@Saritha, to your question, yes, we need to update 'Domanda 1 di 7', as we've been given a more accurate translation. @jijopulikkottil No, there will be no need to adjust the app store info & app display name. 

Thanks again for your work on this, I'll provide updates as they come in.

**Comment by sarithapillai8** [comment]
@ertjlane 
>I've attached some small updates for the Italian translation here.
[Italian Translation Update-12-13-22.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10219412/Italian.Translation.Update-12-13-22.xlsx) 
we need to update 'Domanda 1 di 7', as we've been given a more accurate translation.

These are updated in dashboard-staging.

**Comment by ertjlane** [comment]
@sarithapillai8 Thanks for the update. I just looked, and it doesn't appear as though 'Select ALL', 'Deselect ALL" and 'Create a user under their own group.' have been implemented. They appeared as English on the dashboard, and when I looked in the Italian translation file, I didn't see them in there.

**Comment by divyav2020** [comment]
[Italian.Translation.Update-12-13-22.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10233815/Italian.Translation.Update-12-13-22.xlsx)
These changes are updated in Dashboard staging

**Comment by ertjlane** [comment]
Great thank you!

**Comment by jijopulikkottil** [comment]
@ertjlane FYI - For mobile app, we have not received any translation. https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/issues/697#issuecomment-1341067160 this is the updated list of texts need to translate

**Comment by sarithapillai8** [comment]
@ertjlane We need to release LAMP-activities to complete the production release procedure. Last release release was done on Nov 10 for that repo.

**Comment by ertjlane** [comment]
I believe that is what was happening, everything appears operating now. I have just set the lamp-activity release.

**Comment by sarithapillai8** [comment]
App side changes are in progress. After QA we can update staging.

**Comment by jijopulikkottil** [comment]
@ertjlane Hope we need to translate the Remote notifications. So the notification **"You have a mindLAMP activity waiting for you: pop the bubbles."** will display in device with French language like : **"Une activité mindLAMP vous attend :pop the bubbles."**
We have to implement this in Mobile and Wearable device.

**Comment by jijopulikkottil** [comment]
@ertjlane In the mobile app, translation should work according to the language selection on the MindLamp login page. is it right (And not according to language selection in Device's settings!.) ?

**Comment by ertjlane** [comment]
@jijopulikkottil To your former question, yes I can confirm the remote notifications should be translated. To your second question, I'm not actually 100% sure, you guys are honestly better equipped to answer that question than I am, given that you are implementing the translations. I imagine you can test it on a device as the translations are rolled into staging. 

I also have updated Danish text here. Let me know if you have any questions regarding it!
[mindLAMP_Danish_corrections_22Dec2022_ASAD_PRESCIENT.docx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10336990/mindLAMP_Danish_corrections_22Dec2022_ASAD_PRESCIENT.docx)

**Comment by jijopulikkottil** [comment]
@Linoy339 To translate remote notification, please provide the activity title as part of push payload.
@sarithapillai8 In mobile app, currently we are getting   "authorizationToken", "identityObject" and "serverAddress" when login. please provide the selected language along with those.

**Comment by jijopulikkottil** [comment]
Thanks @sarithapillai8 , but we don't need it. 
As we are using the translated text mostly in Privacy permissions, we think it is okay to display in user's selected language (in device settings). Also dynamically setting the language will lead to restart the application because of info plist texts.

**Comment by Linoy339** [comment]
@jijopulikkottil . The changes has been done in `LAMP-worker` . Now, the payload will have the field `title`, so that you can use it for language translation. Its available in staging

**Comment by jijopulikkottil** [comment]
@ertjlane please inform once you update the staging.

**Comment by ertjlane** [comment]
I believe I have done so! I simply went and made a release in 'lamp-worker'!

**Comment by jijopulikkottil** [comment]
@ertjlane the staging server not seems to be updated yet. @Linoy339 can you confirm

**Comment by ertjlane** [comment]
Am I misunderstanding something about the procedure? I went and made a release in 'lamp-worker'. It sometimes takes a while for a release to go through. Again, please let me know if I'm missing something.

**Comment by jijopulikkottil** [comment]
All merged changes are available on staging server now. So the push notification translations are completed. We are testing backward compatibility.

**Comment by sijitg** [comment]
Android app Open testing version is live in Playstore

**Comment by ertjlane** [comment]
This is great to hear. As it happens, I've received some updated text from the German site (attached here). So these should actually be implemented before we move into the testing phase. For what it's worth I only just received these today, otherwise we wouldn't have to divert course.

German mobile app translation
[mindLAMP2_MobileApp_Localization_Translation.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10502854/mindLAMP2_MobileApp_Localization_Translation.xlsx)
This screenshot is for the German dashboard text, a small modification. 
<img width="596" alt="Screen Shot 2023-01-25 at 2 23 37 PM (2)" src="https://user-images.githubusercontent.com/104088715/214668029-118f3f47-ff97-4133-be72-34478647d120.png">

**Comment by ertjlane** [comment]
As a follow up, I have a review of the Chinese translation. I understand you already received some feedback on the Chinese (back in December). But it looks as though there are still some missing materials on the staging dashboard. I've shared the document including the feedback here. Some of it HAS already been addressed, so please look through it, compare to the staging dashboard and address the pieces that are actually missing. I've also included the Chinese mobile app translations provided (which I just received).
Updated mobile app
[mindLAMP2_MobileApp_Localization (5).xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10512613/mindLAMP2_MobileApp_Localization.5.xlsx)

Chinese Translations update 
[Translation Check - Chinese (1).docx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/10512617/Translation.Check.-.Chinese.1.docx)

Finally, there might be some error affecting the dashboard or staging dashboard. Across multiple languages (ex: Chinese, Korean, German), when going into the notification schedule, the translations are there, but the buttons to modify the translation schedule are gone, and nothing can be selected, as shown in the screenshot below.
<img width="956" alt="Screen Shot 2023-01-26 at 2 05 28 PM (2)" src="https://user-images.githubusercontent.com/104088715/214929925-1892f849-9c1f-43b5-9cb4-2e119b245dfd.png">

Thank you

**Comment by sarithapillai8** [comment]
@ertjlane 
1 = Uberhaupt nicht; 7 = sehr 
I could not find this text in our translation file. This can be a dynamic text created from survey. If you can provide correct english version of this we can add that and can check if that works.

Regarding schedule update issue, could you please confirm there are any existing schedule? In the screenshot there I can see the add button. Could you please confirm?

**Comment by ertjlane** [comment]
@sarithapillai8 
Thank you for updating the dashboard, it looks good!

Regarding the German translation - for every question in every survey on mindLAMP there is a counter noting the number of question out of the total questions, as can be seen in the screenshot below. I might be misunderstanding what you mean by 'dynamic text created from a survey', but just so we're both on the same page, the text is not created by a user per se (as a survey question would be), but it is, as you say, dynamic, the numbers changing depending on the question value. So the numbers should still be able to change when this is implemented. 
<img width="1167" alt="Screen Shot 2023-01-31 at 10 08 13 AM (2)" src="https://user-images.githubusercontent.com/104088715/215798806-950d10b1-64ab-4bd5-be2f-6580b5c1647b.png">

Finally, you're very right - I did not realize the icons did not show up when there was no schedule. Good call.

**Comment by sarithapillai8** [comment]
@ertjlane 
The request was to convert this text  "1 being Uberhaupt nicht; 7 being sehr"  to this one "1 = Uberhaupt nicht; 7 = sehr". This comes for the slider type question as shown below. Not the case you are pointing. This one comes from the question answer option description.  
![image](https://user-images.githubusercontent.com/63710347/216000263-bdd911fc-abc7-4daa-a1b2-ecd6d1144262.png)

**Comment by ertjlane** [comment]
Ah, you're right! Then I don't actually think you have to do anything with it, it is up to someone who wants to use that language to upload or write it into the dashboard themselves correct?

**Comment by sarithapillai8** [comment]
@ertjlane 
You can use the language text while creating the survey.

**Comment by ertjlane** [comment]
@sarithapillai8 Bravo. 

General: As a note while we continue QA testing, we are still awaiting feedback on the Spanish translation. However, I think proceeding with QA testing is still appropriate, with the knowledge that the translations may have to be updated every now and then, even after this issue is formally closed.

**Comment by jijopulikkottil** [comment]
@ertjlane German and Chinese (simplified and traditional) are completed. iOS code has been pushed and the Version 2023.2.1 available for appstore connect users.
Pending are Spanish, Italian and Hindi translations.

**Comment by sijitg** [comment]
@ertjlane German and Chinese (simplified and traditional) are completed in android mobile app .   Code has been pushed and the Version 2023.2.1 available for beta testing. We shall also do post release testing and update .

**Comment by ertjlane** [comment]
Great, thank you all. 

@jijopulikkottil, we are not waiting on Italian & Hindi translations. I have sent you the original Italian translation, and then later feedback, which can be found in this chain. Hindi is not being changed as part of this particular project.

**Comment by jijopulikkottil** [comment]
@ertjlane We couldn't find Italian translation for mobile app in this thread. Found Italian for web [here](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/issues/697#issuecomment-1348797228)

**Comment by divyav2020** [comment]
We have completed post release testing of LAMP beta application.  German and Chinese (simplified and traditional) are updated

**Comment by ertjlane** [comment]
Hello. I've attached her two text files (which can be very easily converted into .json files, the formatting should be correct), along with an excel file for the mobile translations. These are for both Traditional Chinese and Simplified Chinese. So please incorporate the simplified chinese as a new language, and update the traditional chinese materials we already have here with these ones. 
[mindLAMP2_MobileApp_Localization_Chinese_Traditional_And_Simplified.xlsx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/11063060/mindLAMP2_MobileApp_Localization_Chinese_Traditional_And_Simplified.xlsx)
[JSON Traditional Chinese.txt](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/11063061/JSON.Traditional.Chinese.txt)
[JSON Simplified Chinese.txt](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/11063062/JSON.Simplified.Chinese.txt)

**Comment by ZCOEngineer** [comment]
Sure @ertjlane

**Comment by ZCOEngineer** [comment]
We have started this on iOS. Will soon start on Android and Web.

**Comment by sarithapillai8** [comment]
We have updated changes in dashboard-staging. Testing is in progress.

**Comment by ZCOEngineer** [comment]
@ertjlane changes and testing complete on both iOS and Android, we shall push to beta and testflight today,

**Comment by sijitg** [comment]
Android beta build submitted  for review.

**Comment by jijopulikkottil** [comment]
iOS build 2023.4.4 (406) is submitted to Test flight. Status is waiting for review.

**Comment by ZCOEngineer** [comment]
We are seeing a strange issue of surveys getting blank when we do an update. We are checking for reasons, shall update here soon.

**Comment by sarithapillai8** [comment]
We have updated the staging with translation updates. Activity build issues still exist, but that does not affect the working of activities.  We will be checking that issue in parallel.

**Comment by sarithapillai8** [comment]
@ertjlane We are still checking on it. Didn't get any reason yet. The only change was done on the translation strings. So it seems some package version issue is there. Currently, the build files are manually updated, so the activities are working fine in staging. The issue is only with auto-build. If you have to release the build to production, it will work. The repo cannot be re-build through the automated process.

**Comment by ertjlane** [comment]
@sarithapillai8  I recently received some feedback from an Italian team. They have included a list of sections that were missing translations, or for which translations for erroneous, and included the correct translation. Along with this they have screenshots of the location of the translation (so there shouldn't be issues with needing the original text). Could you implement these updates? Thanks. 
[MindLAMP_Italy.zip](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/12006234/MindLAMP_Italy.zip)

**Comment by sarithapillai8** [comment]
We can work on this after priority items.

**Comment by sarithapillai8** [comment]
@ertjlane In the document it is asked to change “7 domande” with “autovalutazione” for Italian. This is common for all languages. Can we change that to 'Self Evaluation' for all languages?

**Comment by sarithapillai8** [comment]
In the given document for Italian translation changes, it is asked to change the text in the activity popup for DBT. It is common for all languages. So we need to change this as it is showing as '7 questions' now and we have conditional questions in DBT, so we can change it to 'Self Evaluation' as suggested in DBT. Please confirm.

**Comment by sarithapillai8** [comment]
1. In general, our question regards the type of graphical presentation of these data that appears not fully helpful for the patient (is it possible to display these data only to researchers?) - Do you have any insight on this?
2. 1. “0.68 minutes taken to complete lever” with “(change time in seconds) impiegati per completare il livello” - Initially we used seconds then changed seconds to minutes as per requirement. So please confirm on this. 
3. Please reply - https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/issues/697#issuecomment-1663292393

**Comment by divyav2020** [comment]
We have updated italian translations (given in MindLAMP_Italy.zip) in dashboard staging. Please confirm.
ACTIVITY 1 - implemented

ACTIVITY 2 - implemented

DASHBOARD and DASHBOARD 2 - implemented

QUESTIONARIO - implemented

PORTALE DATI (Portale Dati) - implemented

DIARY-1 - implemented except “800 max characters” with “Massimo 800 caratteri”

DIARY-2 - implemented

DBT CARD - START - implemented

DBT CARD 2 - implemented

DBT CARD 3 - implemented

DBT CARD 4 - implemented

DBT CARD 5 - implemented

DBT CARD 6 - implemented

DBT CARD 7 - implemented

PALLONCINO (BART TEST) - implemented

SCRATCH-01 - implemented

RESPIRAZIONE-01 - implemented

BUBBLE 1 - implemented except
1. “0.68 minutes taken to complete lever” with “(change time in seconds) impiegati per completare il livello”
2. In general, our question regards the type of graphical presentation of these data that appears not fully helpful for the patient (is it possible to display these data only to researchers?) 

BUBBLE 2 - implemented

BUBBLE 2 bits - implemented

BUBBLE 3 - implemented

BUBBLE 4 - implemented

BUBBLE 8 - implemented

**Comment by ertjlane** [comment]
Hello all,
Yes, confirming that you can go ahead with the changes, both for the DBT card and the shift from seconds to minutes.

**Comment by sarithapillai8** [comment]
@ertjlane Updated in staging. Please check.

**Comment by sarithapillai8** [comment]
@ertjlane We can start work on this using the document attached.

**Comment by ertjlane** [comment]
Great, thank you! It will be a bit tricky, because hardcoded and softcoded materials are somewhat mixed in together. For example, there are a number of surveys, the titles and questions of which are softcoded (ergo not your responsibility to change). However, because those surveys have likert scale responses, we DO need to actual hardcoded response options to be translated in mindLAMP. So it will have to be looked through carefully on a case by case basis. Please let me know if you have any questions.

**Comment by sarithapillai8** [comment]
@ertjlane We have updated almost all the changes. But in SymbolDigitSubstitution, the translations are not working from dashboard. But when running the activity alone, we can see the changes. So we are trying to fix that. Otherwise all other updates are ready in staging.

**Comment by ertjlane** [comment]
@sarithapillai8 Thanks for your hard work on this, I just reviewed the translations. I did notice that 5 strings were not translated in different places, for which I'm attaching an excel sheet. They should be simple fixes. 
[Hindi Translation Fixes(Sheet1).csv](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/14774922/Hindi.Translation.Fixes.Sheet1.csv)

**Comment by ertjlane** [comment]
Hello @sarithapillai8 Just following up on this.

**Comment by sarithapillai8** [comment]
@ertjlane The bugs reported above are fixed and updated in staging. But we still have some weird issues in the symbol digit game, there are some strings still not showing translated even though we are using similar code like other activities. We are working on that.

**Comment by ertjlane** [comment]
@sarithapillai8 Thank you for the update, and getting through most of the fixes quickly. Understood regarding the symbol digit game. 

I have attached a new set of updates our collaborator sent (sections of the translation previously in English that they had not provided translations for), which are highlighted in yellow below. 

[Sharp2_Game_Sectionassess_Hindi_translated_Revised-01-04-2024.docx](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/files/14852597/Sharp2_Game_Sectionassess_Hindi_translated_Revised-01-04-2024.docx)

**Comment by ertjlane** [comment]
I just looked, and it appears on my end as though most of the translations for the symbol-digit game are working in staging. I did notice that some of the english text in the translation file does not exactly match the english text on the dashboard. Ex: in some of the instructions for the game, there is not a space between two words in the English text on the dashboard, but there is a space in the translation file in lamp/activities. Could this be causing the issues you are experiencing?

**Comment by sarithapillai8** [comment]
@ertjlane Almost all the translations in symbol-digit game are working now. We can integrate new changes and will do detailed testing and let you know.

**Comment by ertjlane** [comment]
@sarithapillai8 Just wanted to check in on this. Thank you for your work testing, do we think these are ready for release?

**Comment by sarithapillai8** [comment]
@ertjlane We just completed testing and the given translations are working now. 
Note:  For survey questions, the researcher can add text in the required language. Given translations are added in the translation file.

**Comment by sarithapillai8** [comment]
@ertjlane Reviewed and merged the PR. Dashboard staging will be updated soon.
